modules/azure/compute.go

```diff
@@ -8,97 +8,392 @@ import (
+func ListVirtualMachinesForResourceGroupE(resourceGroupName string, subscriptionID string) ([]string, error) {
+	var vmDetails []string
+
+	vmClient, err := GetVirtualMachineClientE(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	vms, err := vmClient.List(context.Background(), resourceGroupName)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, v := range vms.Values() {
+		vmDetails = append(vmDetails, *v.Name)
+	}
+	return vmDetails, nil
+}
+
+// GetVirtualMachinesForResourceGroup gets all Virtual Machine objects in the specified Resource Group. Each
+// VM Object represents the entire set of VM compute properties accessible by using the VM name as the map key.
+// This function would fail the test if there is an error.
+func GetVirtualMachinesForResourceGroup(t testing.TestingT, resGroupName string, subscriptionID string) map[string]compute.VirtualMachineProperties {
+	vms, err := GetVirtualMachinesForResourceGroupE(resGroupName, subscriptionID)
+	require.NoError(t, err)
+	return vms
+}
+
+// GetVirtualMachinesForResourceGroupE gets all Virtual Machine objects in the specified Resource Group. Each
+// VM Object represents the entire set of VM compute properties accessible by using the VM name as the map key.
+func GetVirtualMachinesForResourceGroupE(resourceGroupName string, subscriptionID string) (map[string]compute.VirtualMachineProperties, error) {
+	vmClient, err := GetVirtualMachineClientE(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	vms, err := vmClient.List(context.Background(), resourceGroupName)
+	if err != nil {
+		return nil, err
+	}
+
+	vmDetails := make(map[string]compute.VirtualMachineProperties, len(vms.Values()))
+	for _, v := range vms.Values() {
+		vmDetails[string(*v.Name)] = *v.VirtualMachineProperties
+	}
+	return vmDetails, nil
+}
+
+// ******************************************************************** //
+// Get VM using Instance and Instance property get, reducing SKD calls
+// ******************************************************************** //
+
+// Instance of the VM
+type Instance struct {
+	*compute.VirtualMachine
+}
+
+// GetVirtualMachineInstance gets a local Virtual Machine instance in the specified Resource Group.
+// This function would fail the test if there is an error.
+func GetVirtualMachineInstance(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) *Instance {
+	vm, err := GetVirtualMachineInstanceE(vmName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+	return vm
+}
+
+// GetVirtualMachineInstanceE gets a local Virtual Machine instance in the specified Resource Group.
+func GetVirtualMachineInstanceE(vmName string, resGroupName string, subscriptionID string) (*Instance, error) {
+	// Get VM Object
+	vm, err := GetVirtualMachineE(vmName, resGroupName, subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Instance{vm}, nil
```

Comment:
I'm a bit confused... This just wraps `GetVirtualMachineE` in a different return type? If so, does this really need two more methods?

Reply:
Removed the two functions and leveraged the existing VM in the test. I kept all the logic in the module at first to have a concrete type assertion in one space but yes, it was redundant at that point.